src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs

```diff
@@ -258,6 +258,12 @@ private void VerifyAuthorizationResult()
                     MsalErrorMessage.NoPromptFailedErrorMessage);
             }
 
+            if (_authorizationResult.Status == AuthorizationStatus.UserCancel)
```

Comment:
Can you look at adding tests for this class? A unit tests should have failed when you added these lines, but it clearly did not since the PR passed. I think it is important to lock this down with unit tests because the exceptions we throw are part of the public API, i.e. if we throw a different exception from one release to the other we have broken customers' handling code.

Reply:
Done.